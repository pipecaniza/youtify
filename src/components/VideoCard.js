import React from 'react';

const VideoCard = ({ video, selectVideo }) => {
  return (    
    <div className="card h-100" >
      <img src={video.thumbnail} className="card-img-top" alt="..." />
      <div className="card-body d-flex flex-column">
        <h5 className="card-title">{video.title}</h5>           
        <button className="btn btn-primary mt-auto btn-block" onClick={() => selectVideo(video.id)}>Play!</button>
      </div>
    </div>
  );
};

export default VideoCard;