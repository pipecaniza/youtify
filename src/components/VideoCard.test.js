import React from 'react'
import renderer from 'react-test-renderer'
import VideoCard from './VideoCard'

it('renders correctly', () => {
  const video = {
    thumbnail: 'url:image',
    title: 'videoTitle',
    id: 1
  };
  const selectVideo = jest.fn();
  const tree = renderer
    .create(<VideoCard video={video} selectVideo={selectVideo} isSignedIn={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});