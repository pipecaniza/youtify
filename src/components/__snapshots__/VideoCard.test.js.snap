// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card h-100"
>
  <img
    alt="..."
    className="card-img-top"
    src="url:image"
  />
  <div
    className="card-body d-flex flex-column"
  >
    <h5
      className="card-title"
    >
      videoTitle
    </h5>
    <div
      className="btn-group"
    >
      <button
        className="btn btn-primary mt-auto btn-block play"
        onClick={[Function]}
      >
        Play!
      </button>
    </div>
  </div>
</div>
`;
